rulesets/java/jpinpoint-java-rules.xml

```diff
@@ -1830,6 +1830,34 @@ public class GoodPerson implements Comparable<Person> {
         </example>
     </rule>
 
+    <rule name="LegacyTimeLibraryUsed"
+          language="java"
+          message="Migrate from the legacy time library to java.time for better performance."
+          class="net.sourceforge.pmd.lang.rule.xpath.XPathRule"
+          externalInfoUrl="https://github.com/jborgers/PMD-jPinpoint-rules/tree/pmd7/docs/JavaCodePerformance.md#uoslc03">
+        <description>Problem: the JodaTime and ThreeTenBp libraries have non-optimal performance and memory usage.
+            Solution: Migrate to java.time for a modern solution with better performance and less memory usage.
+        (jpinpoint-rules)</description>
+        <priority>2</priority>
+        <properties>
+            <property name="tags" value="jpinpoint-rule,performance,sustainability-medium,cpu,memory" type="String" description="classification"/>
+            <property name="xpath">
+                <value><![CDATA[
+//ImportDeclaration[starts-with(@PackageName, 'org.joda.time')][1]
```

Comment:
Considered matching on line where used instead?

Reply:
I considered that, yet, the typeIs is more brittle and I assume taking more resources. This is more simple, also the rule expression.